packages/camera/camera_avfoundation/ios/Classes/CameraPlugin.m

```diff
@@ -41,37 +41,34 @@ + (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar {
 
 - (instancetype)initWithRegistry:(NSObject<FlutterTextureRegistry> *)registry
                        messenger:(NSObject<FlutterBinaryMessenger> *)messenger {
+  return
+      [self initWithRegistry:registry
+                   messenger:messenger
+                   globalAPI:[[FCPCameraGlobalEventApi alloc] initWithBinaryMessenger:messenger]];
+}
+
+- (instancetype)initWithRegistry:(NSObject<FlutterTextureRegistry> *)registry
+                       messenger:(NSObject<FlutterBinaryMessenger> *)messenger
+                       globalAPI:(FCPCameraGlobalEventApi *)globalAPI {
   self = [super init];
   NSAssert(self, @"super init cannot be nil");
   _registry = [[FLTThreadSafeTextureRegistry alloc] initWithTextureRegistry:registry];
   _messenger = messenger;
+  _globalEventAPI = globalAPI;
   _captureSessionQueue = dispatch_queue_create("io.flutter.camera.captureSessionQueue", NULL);
   dispatch_queue_set_specific(_captureSessionQueue, FLTCaptureSessionQueueSpecific,
                               (void *)FLTCaptureSessionQueueSpecific, NULL);
 
-  [self initDeviceEventMethodChannel];
-  [self startOrientationListener];
-  return self;
-}
-
-- (void)initDeviceEventMethodChannel {
-  FlutterMethodChannel *methodChannel = [FlutterMethodChannel
-      methodChannelWithName:@"plugins.flutter.io/camera_avfoundation/fromPlatform"
-            binaryMessenger:_messenger];
-  _deviceEventMethodChannel =
-      [[FLTThreadSafeMethodChannel alloc] initWithMethodChannel:methodChannel];
-}
-
-- (void)detachFromEngineForRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar {
-  [UIDevice.currentDevice endGeneratingDeviceOrientationNotifications];
-}
-
-- (void)startOrientationListener {
```

Comment:
I inlined this as a drive-by cleanup because it was only being called from one place, and that was `init`, which shouldn't be calling methods.